source/distributed/solution_transfer.cc

```diff
@@ -207,26 +229,97 @@ namespace parallel
     SolutionTransfer<dim, VectorType, DoFHandlerType>::pack_callback(
       const typename Triangulation<dim, DoFHandlerType::space_dimension>::
         cell_iterator &cell_,
-      const typename Triangulation<dim, DoFHandlerType::space_dimension>::
-        CellStatus /*status*/)
+      const typename Triangulation<dim,
+                                   DoFHandlerType::space_dimension>::CellStatus
+        status)
     {
       typename DoFHandlerType::cell_iterator cell(*cell_, dof_handler);
 
       // create buffer for each individual object
       std::vector<::dealii::Vector<typename VectorType::value_type>> dofvalues(
         input_vectors.size());
 
-      auto cit_input = input_vectors.cbegin();
-      auto it_output = dofvalues.begin();
-      for (; cit_input != input_vectors.cend(); ++cit_input, ++it_output)
+      if (has_hp_dof_handler)
         {
-          it_output->reinit(cell->get_fe().dofs_per_cell);
-          cell->get_interpolated_dof_values(*(*cit_input), *it_output);
+          unsigned int dofs_per_cell = numbers::invalid_unsigned_int;
+          unsigned int fe_index      = numbers::invalid_unsigned_int;
+
+          switch (status)
+            {
+              case parallel::distributed::Triangulation<
+                dim,
+                DoFHandlerType::space_dimension>::CELL_PERSIST:
+              case parallel::distributed::Triangulation<
+                dim,
+                DoFHandlerType::space_dimension>::CELL_REFINE:
+                dofs_per_cell = cell->get_fe().dofs_per_cell;
+                fe_index      = cell->active_fe_index();
+                break;
```

Comment:
clang-format indents this awkwardly, making it impossible to see what is the label and what is code. can you wrap the code into curly braces to make this easier to read?